packages/edit-site/src/components/page-templates/dataviews-templates.js

```diff
@@ -0,0 +1,199 @@
+import { TEMPLATE_POST_TYPE } from '../../utils/constants';
+import { DataViews } from '../dataviews';
+import {
+	useResetTemplateAction,
+	deleteTemplateAction,
+	renameTemplateAction,
+} from './template-actions';
+
+const EMPTY_ARRAY = [];
+
+const DEFAULT_VIEW = {
+	type: 'list',
+	search: '',
+	page: 1,
+	perPage: 20,
+	// All fields are visible by default, so it's
+	// better to keep track of the hidden ones.
+	hiddenFields: [],
+	layout: {},
+};
+
+function normalizeSearchInput( input = '' ) {
+	return removeAccents( input.trim().toLowerCase() );
+}
+
+export default function DataviewsTemplates() {
+	const [ view, setView ] = useState( DEFAULT_VIEW );
+	const { records: allTemplates, isResolving: isLoadingData } =
+		useEntityRecords( 'postType', TEMPLATE_POST_TYPE, {
+			per_page: -1,
+		} );
+	const { shownTemplates, paginationInfo } = useMemo( () => {
+		if ( ! allTemplates ) {
+			return {
+				shownTemplates: EMPTY_ARRAY,
+				paginationInfo: { totalItems: 0, totalPages: 0 },
+			};
+		}
+		let filteredTemplates = [ ...allTemplates ];
+		// Handle global search.
+		if ( view.search ) {
+			const normalizedSearch = normalizeSearchInput( view.search );
+			filteredTemplates = filteredTemplates.filter( ( item ) => {
+				const title = item.title?.rendered || item.slug;
+				return (
+					normalizeSearchInput( title ).includes(
+						normalizedSearch
+					) ||
+					normalizeSearchInput( item.description ).includes(
+						normalizedSearch
+					)
+				);
+			} );
+		}
+		// Handle sorting.
+		// TODO: Explore how this can be more dynamic..
+		if ( view.sort ) {
+			if ( view.sort.field === 'title' ) {
+				filteredTemplates.sort( ( a, b ) => {
+					const titleA = a.title?.rendered || a.slug;
+					const titleB = b.title?.rendered || b.slug;
+					return view.sort.direction === 'asc'
+						? titleA.localeCompare( titleB )
+						: titleB.localeCompare( titleA );
+				} );
+			}
+		}
+		// Handle pagination.
+		const start = ( view.page - 1 ) * view.perPage;
+		const totalItems = filteredTemplates?.length || 0;
+		filteredTemplates = filteredTemplates?.slice(
+			start,
+			start + view.perPage
+		);
+		return {
+			shownTemplates: filteredTemplates,
+			paginationInfo: {
+				totalItems,
+				totalPages: Math.ceil( totalItems / view.perPage ),
+			},
+		};
+	}, [ allTemplates, view ] );
+	const fields = useMemo(
+		() => [
+			{
+				header: __( 'Template' ),
+				id: 'title',
+				getValue: ( { item } ) => item.title?.rendered || item.slug,
+				render: ( { item } ) => {
+					return (
+						<VStack spacing={ 1 }>
+							<Heading as="h3" level={ 5 }>
+								<Link
+									params={ {
+										postId: item.id,
+										postType: item.type,
+										canvas: 'edit',
+									} }
+								>
+									{ decodeEntities(
+										item.title?.rendered || item.slug
+									) || __( '(no title)' ) }
+								</Link>
+							</Heading>
+						</VStack>
+					);
+				},
+				maxWidth: 400,
+				sortingFn: 'alphanumeric',
```

Comment:
```suggestion
```

I was looking at what `sortingFn` does and realized it does nothing. Fix for pages at https://github.com/WordPress/gutenberg/pull/55989